### PR TITLE
Refactor dwave.optimization.generators

### PR DESCRIPTION
**Pull request by arcondello**
* Clean up pep8 errors and whitespace
* Remove redundant tests in tests/test_examples.py
* Refactor input argument checking/coercion
* Slightly generalize some of the generators

### CODE REVIEW

File: dwave/optimization/generators.py [hunk 1]
```diff
@@ -17,120 +17,64 @@
...
-    for the error message to make sense.
-    """
-    for argname, array in kwargs.items():
-        try:
-            array = np.atleast_2d(np.asarray(array))
-        except (ValueError, TypeError):
-            raise ValueError(f"`{argname}` must be a square 2D array-like of non-negative numbers")
+        # we do "size up"
+        while array.ndim < ndim:
+            array = array[np.newaxis]
```
Comment (JoelPasvolsky) [inline dwave/optimization/generators.py:66]: I'm not quite sure what is intended with this clever trick but I doubt it is this:

```
model = job_shop_scheduling(times=[1, 2, 3, 4], machines=[4, 3, 2, 1])
ValueError: arrays must have the same shape
```

I think the code will be clearer if, for a requirement on square matrices, you check that `array.shape[0] == array.shape[1]` and don't upsize.

**Review by JoelPasvolsky — COMMENTED** [review]
That's a lot of trailing spaces you cleaned up.

File: dwave/optimization/generators.py (hunk 1 above)
Comment (arcondello) [reply dwave/optimization/generators.py:66]: That error was present in the original code. Turns out it fails on `main` for every non-square times/machines. Will fix.

Re the dimension increase, I am following the pattern of [np.atleast_2d()](https://numpy.org/doc/stable/reference/generated/numpy.atleast_2d.html), which is typical in this sort of place.

Comment (arcondello) [reply dwave/optimization/generators.py:66]: Ah, that's a different bug. Will fix that one too.

Comment (arcondello) [reply dwave/optimization/generators.py:66]: https://github.com/dwavesystems/dwave-optimization/issues/43

**Review by JoelPasvolsky — APPROVED** [review]

**Pull request merged**